src/proc_diskstats.c

```diff
@@ -19,16 +22,23 @@
 
 #define RRD_TYPE_DISK "disk"
 
+#define DISK_TYPE_UNKNOWN  -1
+#define DISK_TYPE_PHYSICAL  1
+#define DISK_TYPE_PARTITION 2
+#define DISK_TYPE_CONTAINER 3
+
 struct disk {
 	unsigned long major;
 	unsigned long minor;
-	int partition_id;       // -1 = this is not a partition
-	char *family;
+	int type;
+	char *mount_point;
 	struct disk *next;
 } *disk_root = NULL;
 
+struct statvfs buff_statvfs;
```

Comment:
please put this inside the function as `static`.
No major harm, just to comply with the rest of the code.